### ISSUE TITLE
Zephyr logging: migrate from using SYS_LOG to LOG infrastructure

### ISSUE THREAD

**Issue by nvlsianpu** [JuulLabs-OSS/mcuboot#350]
It is planed that new zephyr logging infrastructure will replace old SYS_LOG module soon.
An example of migration is here: https://github.com/zephyrproject-rtos/zephyr/pull/10546

Related to [Zephyr Logging Subsystem Enhancements](https://github.com/zephyrproject-rtos/zephyr/issues/6186)

**Comment by nvlsianpu** [comment]
cc @nordic-krch

**Comment by utzig** [comment]
I assume this is the reason for all those warnings now:

```
warning: Macro is deprecated
         BOOT_LOG_ERR(...)
                   ^~~~~~~~~~~~~~~~~~~~~
```
Correct?

**Comment by arn-soft** [comment]
@utzig yes, that's the reason of that warning. moving on new LOG infrastructure will potentially solve the issue.

**Comment by nvlsianpu** [comment]
fixed by #360
